### ISSUE TITLE
Support for emoji skin tone variations

### ISSUE THREAD

**Issue by StephenTangCook** [themashcodee/slack-blocks-to-jsx#16]
Emojis can have modifier sequences, most commonly used for skin-tone variables. There can be up to six color variations (with the first one assumed to be the default). Here's an example for thumbs-up:
```
  "thumbsup": "👍",
  "thumbsup::skin-tone-2": "👍🏻",
  "thumbsup::skin-tone-3": "👍🏼",
  "thumbsup::skin-tone-4": "👍🏽",
  "thumbsup::skin-tone-5": "👍🏾",
  "thumbsup::skin-tone-6": "👍🏿",
```

Slack supports changing your default skin tone for emojis, so a full emoji with modifier sequence could appear in a message.

**Comment by StephenTangCook** [comment]
In one of my projects we have a script that just creates a static mapping of all possible emoji variation shortnames from `emoji-datasource` (which gets its data from `emoji-data`, the same emoji library Slack uses). You're free to reuse the logic if it helps!

Essentially we have the scripts:
```js
  "scripts": {
    "build": "tsup src/index.ts",
    "copy-emoji-data": "cp node_modules/emoji-datasource/emoji.json dist/emoji.json",
    "create-emoji-mapping": "pnpm copy-emoji-data && ts-node src/create-emoji-mapping.ts"
  },
```

And the `create-emoji-mapping.ts`:
```js
import EmojiConvertor from 'emoji-js';
import * as fs from 'fs-extra';
import * as path from 'path';

type EmojiEntry = {
  name: string;
  unified: string;
  non_qualified: string | null;
  docomo: string | null;
  au: string | null;
  softbank: string | null;
  google: string | null;
  image: string;
  sheet_x: number;
  sheet_y: number;
  short_name: string;
  short_names: string[];
  text: string | null;
  texts: string[] | null;
  category: string;
  subcategory: string;
  sort_order: number;
  added_in: string;
  has_img_apple: boolean;
  has_img_google: boolean;
  has_img_twitter: boolean;
  has_img_facebook: boolean;
  skin_variations?: Record<string, SkinVariation>;
};

type SkinVariation = {
  unified: string;
  non_qualified: string | null;
  image: string;
  sheet_x: number;
  sheet_y: number;
  added_in: string;
  has_img_apple: boolean;
  has_img_google: boolean;
  has_img_twitter: boolean;
  has_img_facebook: boolean;
};

type EmojiData = EmojiEntry[];

type EmojiShortnameMapping = Record<string, string>;

console.log('Creating emoji short_name mapping...');

// Get the emoji source data
const emojiJsonPath = path.resolve('dist', 'emoji.json');
console.log(`Loading emoji data file at '${emojiJsonPath}'...`);
const emojiData: EmojiData = fs.readJsonSync(emojiJsonPath);

// Get the emoji converter
const emojiConverter = new EmojiConvertor();
emojiConverter.replace_mode = 'unified';

// Get the skintone variations (e.g. "1F3FE" => "skin-tone-5")
const skintoneVariationsMapping: Record<string, string> = {};
emojiData
  .filter(
    (emojiEntry: EmojiEntry) =>
      emojiEntry.category && emojiEntry.subcategory === 'skin-tone'
  )
  .map((skintoneEntry: EmojiEntry) => {
    skintoneVariationsMapping[skintoneEntry.unified] = skintoneEntry.short_name;
  });

// Iterate through the emoji data and create a mapping
const emojiShortnameMapping: EmojiShortnameMapping = {};
emojiData.forEach((emojiEntry: EmojiEntry) => {
  emojiEntry.short_names.forEach((shortname: string) => {
    // start with default shortname (no modifier) and add variations as needed
    const allShortnameVariations: string[] = [shortname];

    // add any skintone variations
    if (emojiEntry.skin_variations) {
      const skinVariationUnifieds: string[] = Object.keys(
        emojiEntry.skin_variations
      );
      skinVariationUnifieds.map((skinVariationUnified) => {
        const skinVariationShortname =
          skintoneVariationsMapping[skinVariationUnified];
        allShortnameVariations.push(`${shortname}::${skinVariationShortname}`);
      });
    }

    // look up the emoji for each shortname variation
    allShortnameVariations.map((shortnameVariation) => {
      const emoji = emojiConverter.replace_colons(`:${shortnameVariation}:`);
      emojiShortnameMapping[shortnameVariation] = emoji;
    });
  });
});

// Save to output file
const outputFile = 'emoji-shortcode-mapping.json';
const outputFilePretty = 'emoji-shortcode-mapping_pretty.json';
fs.writeJsonSync('src/' + outputFile, emojiShortnameMapping);
fs.writeJsonSync('src/' + outputFilePretty, emojiShortnameMapping, {
  spaces: 2
});
```

**Comment by StephenTangCook** [comment]
Note I just updated the script content with some skin tone variation bug fixes, in particular when there are multiple variations for compound emojis (e.g.  `two_women_holding_hands::skin-tone-2-5`: "👩🏻‍🤝‍👩🏾").

**Comment by themashcodee** [comment]
@StephenTangCook Thank you so much for the file, this is super helpful.

**Comment by themashcodee** [comment]
@StephenTangCook resolved in [v0.3.8](https://github.com/themashcodee/slack-blocks-to-jsx/releases/tag/v0.3.8)

**Comment by StephenTangCook** [comment]
Awesome! I swear I'll open source the emoji list conversation in an npm one day when I find the time! 😭
